[BUG] regex_test failed mismatched cpu and gpu values in UT and IT

**Describe the bug**
Below failed unit tests
```
[2024-10-13T00:16:05.623Z] - replace_replace - ?, *, +, and {0, n} repetitions *** FAILED ***
[2024-10-13T00:16:05.623Z]   javaPattern[3]=D{0,}, cudfPattern=D{0,}, input='SS', cpu=_RE\PLACE_S_RE\PLACE_S_RE\PLACE_, gpu=SS (RegularExpressionTranspilerSuite.scala:985)
...
[2024-10-13T00:56:01.073Z] - AST fuzz test - regexp_find *** FAILED ***
[2024-10-13T00:56:01.073Z]   javaPattern[89]=(?:"{0,}), cudfPattern=(?:"{0,}), input='', cpu=true, gpu=false (RegularExpressionTranspilerSuite.scala:958)
[2024-10-13T00:56:02.957Z] - AST fuzz test - regexp_replace *** FAILED ***
[2024-10-13T00:56:02.957Z]   javaPattern[87]=(?:"{0,}), cudfPattern=(?:"{0,}), input='', cpu=_RE\PLACE_, gpu= (RegularExpressionTranspilerSuite.scala:985)
[2024-10-13T00:56:02.957Z] - AST fuzz test - regexp_find - full unicode input !!! CANCELED !!!
[2024-10-13T00:56:02.957Z]   RegularExpressionTranspilerSuite.this.isUnicodeEnabled() was false (RegularExpressionTranspilerSuite.scala:724)
[2024-10-13T00:56:02.957Z] - AST fuzz test - regexp_replace - full unicode input !!! CANCELED !!!
[2024-10-13T00:56:02.957Z]   RegularExpressionTranspilerSuite.this.isUnicodeEnabled() was false (RegularExpressionTranspilerSuite.scala:730)
[2024-10-13T00:56:04.841Z] - AST fuzz test - regexp_find - anchor focused *** FAILED ***
[2024-10-13T00:56:04.841Z]   javaPattern[96]=\07{0,}, cudfPattern=\07{0,}, input='', cpu=true, gpu=false (RegularExpressionTranspilerSuite.scala:958)
...
[2024-10-13T00:56:05.095Z] - regexp_split - repetition with {0,n}, or {0,} *** FAILED ***
[2024-10-13T00:56:05.095Z]   string_split java pattern=ba{0,} cudfPattern=ba{0,} isRegex=true data=bab limit=-2147483648 
[2024-10-13T00:56:05.096Z]   CPU [3]: , ,  
[2024-10-13T00:56:05.096Z]   GPU [2]: , b (RegularExpressionTranspilerSuite.scala:894)
...
[2024-10-13T00:56:23.375Z] - string split fuzz *** FAILED ***
[2024-10-13T00:56:23.375Z]   string_split java pattern=\0514{0,} cudfPattern=\)4{0,} isRegex=true data=b31!@a++) limit=-2
[2024-10-13T00:56:23.375Z]   CPU [2]: b31!@a++,  
[2024-10-13T00:56:23.375Z]   GPU [1]: b31!@a++) (RegularExpressionTranspilerSuite.scala:894)
[2024-10-13T00:56:49.866Z] - string split fuzz - anchor focused *** FAILED ***
[2024-10-13T00:56:49.866Z]   string_split java pattern=\r\n{0,1} cudfPattern=\r\n{0,1} isRegex=true data=ca\rb limit=-1 
[2024-10-13T00:56:49.866Z]   CPU [2]: ca, b 
[2024-10-13T00:56:49.866Z]   GPU [1]: ca\rb (RegularExpressionTranspilerSuite.scala:894)
...
[2024-10-13T00:19:08.569Z] *** 7 TESTS FAILED ***
```


Below failed integration tests
```
src.main.python.regexp_test.test_re_replace_repetition[DATAGEN_SEED=1728771476, TZ=UTC]
src.main.python.regexp_test.test_regexp_replace_digit[DATAGEN_SEED=1728771476, TZ=UTC]
src.main.python.regexp_test.test_regexp_memory_ok[DATAGEN_SEED=1728771476, TZ=UTC, INJECT_OOM]
```

different CPU vs GPU values
```
AssertionError: GPU and CPU string values are different at [0, 'regexp_replace(a, A{0,}, PROD, 1)']
AssertionError: GPU and CPU string values are different at [0, 'regexp_replace(a, [a\\d]{0,2}, x, 1)']
AssertionError: GPU and CPU boolean values are different at [0, 'RLIKE(a, (?:ab){0,3})']
```


**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
Pass the tests

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/regexp_test.py
+++ b/integration_tests/src/main/python/regexp_test.py
@@ -284,7 +284,6 @@ def test_re_replace():
 
 # We have shims to support empty strings for zero-repetition patterns
 # See https://github.com/NVIDIA/spark-rapids/issues/5456
-@pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/11600")
 def test_re_replace_repetition():
     gen = mk_str_gen('.{0,5}TEST[\ud720 A]{0,5}')
     assert_gpu_and_cpu_are_equal_collect(
@@ -699,7 +698,6 @@ def test_regexp_octal_digits():
             ),
         conf=_regexp_conf)
 
-@pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/11600")
 def test_regexp_replace_digit():
     gen = mk_str_gen('[a-z]{0,2}[0-9]{0,2}') \
         .with_special_case('䤫畍킱곂⬡❽ࢅ獰᳌蛫青') \
@@ -1078,7 +1076,6 @@ def test_regexp_memory_fallback():
         }
     )
 
-@pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/11600")
 def test_regexp_memory_ok():
     gen = StringGen('test')
     assert_gpu_and_cpu_are_equal_collect(

--- a/tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala
+++ b/tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala
@@ -418,8 +418,7 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
 
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("replace_replace - ?, *, +, and {0, n} repetitions") {
+  test("replace_replace - ?, *, +, and {0, n} repetitions") {
     val patterns = Seq("D?", "D*", "D+", "D{0,}", "D{0,1}", "D{0,5}", "[1a-zA-Z]{0,}",
         "[1a-zA-Z]{0,2}", "A+")
     val inputs = Seq("SS", "DD", "SDSDSDS", "DDDD", "DDDDDD", "ABCDEFG")
@@ -711,27 +710,23 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
     }
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("AST fuzz test - regexp_find") {
+  test("AST fuzz test - regexp_find") {
     doAstFuzzTest(Some(REGEXP_LIMITED_CHARS_FIND), REGEXP_LIMITED_CHARS_FIND,
       RegexFindMode)
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("AST fuzz test - regexp_replace") {
+  test("AST fuzz test - regexp_replace") {
     doAstFuzzTest(Some(REGEXP_LIMITED_CHARS_REPLACE), REGEXP_LIMITED_CHARS_REPLACE,
       RegexReplaceMode)
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("AST fuzz test - regexp_find - full unicode input") {
+  test("AST fuzz test - regexp_find - full unicode input") {
     assume(isUnicodeEnabled())
     doAstFuzzTest(None, REGEXP_LIMITED_CHARS_REPLACE,
       RegexFindMode)
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("AST fuzz test - regexp_replace - full unicode input") {
+  test("AST fuzz test - regexp_replace - full unicode input") {
     assume(isUnicodeEnabled())
     doAstFuzzTest(None, REGEXP_LIMITED_CHARS_REPLACE,
       RegexReplaceMode)
@@ -741,8 +736,7 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
     Charset.defaultCharset().name() == "UTF-8"
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("AST fuzz test - regexp_find - anchor focused") {
+  test("AST fuzz test - regexp_find - anchor focused") {
     doAstFuzzTest(validDataChars = Some("\r\nabc"),
       validPatternChars = "^$\\AZz\r\n()[]-", mode = RegexFindMode)
   }
@@ -784,8 +778,7 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
     }
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("regexp_split - repetition with {0,n}, or {0,}") {
+  test("regexp_split - repetition with {0,n}, or {0,}") {
     // see https://github.com/NVIDIA/spark-rapids/issues/6958
     val patterns = Set("ba{0,}", raw"a\02{0,}", "ba{0,2}", raw"b\02{0,10}")
     val data = Seq("abaa", "baba", "ba\u0002b", "ab\u0002b\u0002a")
@@ -839,8 +832,7 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
     }
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("string split fuzz") {
+  test("string split fuzz") {
     val (data, patterns) = generateDataAndPatterns(Some(REGEXP_LIMITED_CHARS_REPLACE),
       REGEXP_LIMITED_CHARS_REPLACE, RegexSplitMode)
     for (limit <- Seq(-2, -1, 2, 5)) {
@@ -848,8 +840,7 @@ class RegularExpressionTranspilerSuite extends AnyFunSuite {
     }
   }
 
-  // Disabling until https://github.com/NVIDIA/spark-rapids/issues/11600 is fixed
-  ignore("string split fuzz - anchor focused") {
+  test("string split fuzz - anchor focused") {
     val (data, patterns) = generateDataAndPatterns(validDataChars = Some("\r\nabc"),
       validPatternChars = "^$\\AZz\r\n()", RegexSplitMode)
     doStringSplitTest(patterns, data, -1)

